EdvClient.js

```diff
@@ -347,40 +347,47 @@ export class EdvClient {
    * Deletes a document from the EDV.
    *
    * @param {Object} options - The options to use.
-   * @param {string} options.id the ID of the document to delete.
+   * @param {string} options.doc the document to delete.
+   * @param {Object} [options.recipients=[]] a set of JWE recipients to encrypt
+   *   the document for; if present, recipients will be added to any existing
+   *   recipients; to remove existing recipients, modify the
+   *   `encryptedDoc.jwe.recipients` field.
    * @param {string} [options.capability=undefined] - The OCAP-LD authorization
    *   capability to use to authorize the invocation of this operation.
    * @param {Object} options.invocationSigner - An API with an
    *   `id` property and a `sign` function for signing a capability invocation.
+   * @param {function} [keyResolver=this.keyResolver] a function that returns
+   *   a Promise that resolves a key ID to a DH public key.
+   * @param {Object} [keyAgreementKey=null] a default KeyAgreementKey API for
+   *   deriving shared KEKs for wrapping content encryption keys.
    *
-   * @return {Promise<Boolean>} resolves to `true` if the document was deleted
-   *   and `false` if it did not exist.
+   * @return {Promise<Boolean>} resolves to `true` when the document was
+   *  deleted.
    */
-  async delete({id, capability, invocationSigner}) {
-    _assertString(id, '"id" must be a string.');
+  async delete({
+    doc, recipients = [], capability, invocationSigner,
+    keyResolver = this.keyResolver, keyAgreementKey = this.keyAgreementKey}) {
+    _assertDocument(doc);
+    _assertString(doc.id, '"id" must be a string.');
     _assertInvocationSigner(invocationSigner);
 
-    const url = this._getDocUrl(id, capability);
-    if(!capability) {
-      capability = this._getRootDocCapability(id);
+    // if no recipients specified, add default
+    if(recipients.length === 0 && keyAgreementKey) {
+      recipients = this._createDefaultRecipients(keyAgreementKey);
     }
+
+    doc.content = {};
+    if(!doc.meta) {
+      doc.meta = {};
+    }
+    doc.meta.deleted = true;
+
     try {
-      // sign HTTP header
-      const headers = await signCapabilityInvocation({
-        url, method: 'delete', headers: this.defaultHeaders,
-        capability, invocationSigner,
-        // TODO: should `delete` be used here as a separate action?
-        capabilityAction: 'write'
+      await this.update({
+        doc, keyResolver, keyAgreementKey, capability, invocationSigner
       });
-      // send request
-      const {httpsAgent} = this;
-      await axios.delete(url, {headers, httpsAgent});
     } catch(e) {
-      const {response = {}} = e;
-      if(response.status === 404) {
-        return false;
-      }
-      throw e;
```

Comment:
It seems like we need to throw some exceptions, right? This will just swallow and ignore them entirely. What's an ok exception here? Or should we always throw?

Reply:
I was unsure of how exactly an error could pop up, since it requires the document to be passed in. In `update` there is a check for `409 status code` which then creates a `Conflict error.`. I think that should be added here, and if you could advise of any other errors I might encounter and how to handle/test for them.

Reply:
I followed the same error convention as `update`.